### ISSUE TITLE
Maybe pick another name?

### ISSUE THREAD

**Issue by mikl** [arendjr/selectivity#1]
This looks like a really great project, kudos.

I couldn't help but wince at the name. Unless you have an agreement with the people behind [select2](https://github.com/select2/select2), I think you should consider picking a different name.

From the standard naming patterns (Windows 7 < Windows 8 < Windows 9), the name could be interpreted as this being a newer, better version of select2, rather than a different project.

Additionally, the current stable version of select2 is 3.x. So there's also a lot of potential for confusion between select2 3.x and select3 1.x. And this will just get worse over time.

So from my perspective (which could, of course, be completely mistaken), I think the classy thing to do was to change the name.

**Comment by arendjr** [comment]
I have actually considered changing the name, mainly because of the confusion argument. Select2 is actually soon to release their v4, so yeah, maybe the name could be better to avoid confusion.

Of course, a name that is a little bit controversial might actually help catch people's attention, so I don't think it was necessarily a bad choice. And while it is indeed a different project, the name was meant to imply it is newer and better than Select2 and it is meant to supersede it. At least for me it has :)

I have announced the project on the Select2 mailling list, and while I haven't really asked for their blessing, there were no objections to the name either. So while I'm not entirely opposed to renaming, I don't think there's much urge either.

Finally there's the question, if I were to rename, what should I rename to? I've given it some thought, but frankly, I haven't come up with anything better than Select3. So I'm leaving this issue open, as I'm open to suggestions :)

**Comment by kevin-brown** [comment]
Coming at this from a different perspective (the name doesn't bother me): you're going to have difficulty when people look for information on Select3 because of how high Select2 ranks.

> if I were to rename, what should I rename to?

I gave this some time as well when I was writing up my reply to your announcement, and I came to the [common conclusion that naming is hard](http://martinfowler.com/bliki/TwoHardThings.html), again.

**Comment by mikl** [comment]
Well, as far as names go, it doesn't really matter if it's really related. BadgerSelect, Select-o-turban, selectaroo. Or call it Bjarne. In my opinion, at least, any name that doesn't look like plagiarism would be better.

**Comment by arendjr** [comment]
I've just release Select3 v1.0 and decided to stick with the name. The Select2 maintainers are fine with it, and fortunately Google is picking up on Select3 too. I'll add a little disclaimer to the migration section though to clarify Select3 is not officially endorsed by Select2.

**Comment by jjt** [comment]
+1 for the name change.

I expected this to be the next iteration of Select2, done by the same core contributors. Naming it Select3 just introduces confusion, and you're piggybacking off of it for the benefit of your project. IMO, it's close to being false advertising.

I don't generally like trademark law, but having clear brands is beneficial. Imagine someone made a smartwatch and named it Apple Watch 2, because they wanted the name to "imply it is newer and better than [the first Apple Watch] and it is meant to supersede it". People would buy it expecting that it came from Apple and is the next iteration of their watch.

This isn't a knock on the quality of your project, just the name.

**Comment by JonAbrams** [comment]
At least include a disclaimer saying that this is a different project, along with a link to Select2.

**Comment by kevin-brown** [comment]
@JonAbrams It's mentioned _all the way_ at the end, in the migration section.

https://github.com/arendjr/select3#migrating-from-select2

**Comment by reflectiveSingleton** [comment]
It needs to not be burried at the end in the migration section. Calling this project select3 is very bad-form IMO and will create confusion.

It needs to be clear at the start of it (closer to the top, more apparent) that this isn't the 'new version' of select2.

The maintainers of select2 may not care but I think the majority of the internet would prefer less ambiguous names for projects they are googling.

**Comment by arendjr** [comment]
The reason I put that notice in the migration section is because I feared offering a migration section at all would imply to users it is indeed an "upgrade path". At the very top is a motivation section (where I do link back to the original section) which I thought would already make clear this is a separate project.

Anyway, apparently there's still a lot of confusion which I honestly hadn't expected (previous announcements hadn't gathered any backlash over this with the sole exception of this reported issue).

I apologize for the confusion and will commit myself to rename the project within a week (still drawing a blank on the alternative name right now...).

**Comment by SemiNormal** [comment]
Suggestions (have not checked if any of these are taken):

- Selective
- Choosy
- Persnickety

**Comment by jjt** [comment]
I think you're making the right move here, and I hope to see this project succeed!

* Optionator
* Selectate
* Opty
* Selecty
* SlickPick
* Select[Friend/Thing/etc]
* [adjective]Select

**Comment by holloway** [comment]
* Selecting.js (name doesn't appear to be taken)
* Cellect.js (name is taken but project hasn't been updated in a year)

**Comment by promogest** [comment]
I vote for changing name. 
i suggest:
prettyselect
easyselect
select-ng
jselect

**Comment by arendjr** [comment]
Thanks for all the suggestions, guys! I think I'm gonna settle on selectivity.js. I will update the project as soon as I can!

**Comment by arendjr** [comment]
Issue solved! Select3 is now officially known as Selectivity.js (or just Selectivity for short).

Unfortunately, the downside is that anyone upgrading will have to do a search/replace for himself as well.
The API is now `$.fn.selectivity()` instead of `$.fn.select3()` and all CSS classes are now prefixed with 'selectivity-'.

(I did consider leaving $.fn.select3() as fallback, but given that all templates and CSS classes have been changed as well, that would only solve half the problem...)
